flexget/components/notify/notifiers/gotify.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import base64
+import datetime
+import re
+
+from flexget import plugin
+from flexget.event import event
+from flexget.config_schema import one_or_more
+from flexget.plugin import PluginWarning
+from flexget.utils.requests import Session as RequestSession, TimedLimiter
+from requests.exceptions import RequestException
+
+plugin_name = 'gotify'
+log = logging.getLogger(plugin_name)
+
+requests = RequestSession(max_retries=3)
+
+gotify_url_pattern = {
+    'type': 'string',
+    'pattern': r'^http(s?)\:\/\/.*\/message$',
+    'error_pattern': 'Gotify URL must begin with http(s) and end with `/message`',
+}
+
+class GotifyNotifier(object):
+    """
+    Example::
+
+    notify:
+      entries:
+        via:
+          - gotify:
+              url: <GOTIFY_SERVER_URL>
+              token: <GOTIFY_TOKEN>
+              priority: <PRIORITY>
+    Configuration parameters are also supported from entries (eg. through set).
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': one_or_more(gotify_url_pattern),
+            'token': {'type': 'string'},
+            'priority': {'type': 'integer', 'default': 4},  
+        },  
+        'required': [
+            'token',
+            'url',
+        ],
+        'error_oneOf': 'One (and only one) of `url`, `token` or `priority` are allowed.',
+        'additionalProperties': False,
+    }
+
+
+    def notify(self, title, message, config):
+        """
+        Send a Gotify notification
+        """
+        priority = config.get('priority')      
+        token = config.get('token')
+        url = f"{config.get('url') + '?token=' + token}"
```

Comment:
I would do this like so:
```python
url = f'{config["url"]}?token={token}'
```
But just now I realized this is actually a request param, so you could just pass it as a params dict to requests.

In fact, I'm not sure why even separate between `notify` and `send_push` here, seems kinda redundant to me. You could just move all the logic into `notify` and then just do something like:
```python
url = config['url']
params = {'token': config['token']}
...
response = requests.post(url, params=params, json=notification)
```

Reply:
I like that idea, will get this simplified.

Reply:
https://github.com/Flexget/Flexget/commit/46a8b002822a6d6aa358b4946e32bd004c4fe5d4